refactor: Discovery & Browser, OK-36161, OK-36275

limichange opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Introduced a dedicated home tab for browser navigation that adapts header displays.
  - The add tab action now instantly creates a home tab, streamlining how users manage tabs.
  - The desktop browser view shows dashboard content when no website is available.
  - Added a new `addBrowserHomeTab` method for enhanced tab management.
  - New optional property `type` added to web tab categorization.
  - Enhanced website handling through a new `useWebSiteHandler` hook.

- **Refactor**
  - Unified and simplified website navigation handling across various views for improved consistency.
  - Transitioned to a hook-based approach for handling website navigation in several components.

- **Chores**
  - Enhanced development tooling by enforcing stricter React hooks usage and updating dependency versions.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

limichange committed:
refactor: Centralize web site handling in Discovery module

limichange commented:
@rabbitcode pause

limichange committed:
Merge branch 'x' into OK-36161

limichange committed:
Merge branch 'x' into OK-36161

limichange committed:
refactor: Remove switchToMultiTabBrowser parameter from web site handling

limichange committed:
Merge branch 'x' into OK-36161

limichange committed:
Merge branch 'x' into OK-36161

limichange committed:
refactor: Simplify multi-tab browser logic in discovery actions

- Removed unused media query dependency and updated logic to determine multi-tab browser behavior based on platform environment.
- Improved readability by consolidating conditions for switching to multi-tab browser.

limichange committed:
refactor: Comment out navigation pushModal for DiscoveryModal in DesktopCustomTabBar

- Temporarily disable the navigation to DiscoveryModal's SearchModal to address ongoing development needs.
- Added a TODO note for future reference.

limichange committed:
feat: Add ESLint rule for React hooks compliance

- Introduced 'react-hooks/rules-of-hooks' rule to enforce proper usage of React hooks.

limichange committed:
chore: Update eslint-plugin-react-hooks to version 4.6.2 in package.json and yarn.lock

limichange committed:
Merge branch 'x' into OK-36161

limichange committed:
feat: Add addBrowserHomeTab action and integrate into DesktopCustomTabBar

- Introduced addBrowserHomeTab method to create a new browser tab with the home page and set it as active.
- Integrated addBrowserHomeTab into DesktopCustomTabBar for direct access when the home tab is activated.

limichange committed:
feat: Enhance web tab actions with type differentiation

- Updated addBlankWebTab and addBrowserHomeTab methods to include a 'type' property for web tabs, distinguishing between 'normal' and 'home' types.
- Modified IWebTab interface to accommodate the new 'type' property for improved semantic clarity.

limichange committed:
feat: Update addBrowserHomeTab to include dynamic tab properties

- Enhanced the addBrowserHomeTab method to generate a unique ID and set default properties for the home tab, improving tab initialization.
- Integrated DashboardContent as a fallback display when no URL is present in the BasicDesktopBrowserContent component.

limichange committed:
feat: Enhance DesktopBrowser with dynamic tab handling

- Added useWebTabDataById hook to retrieve active tab data, enabling dynamic behavior based on tab type.
- Updated header title and right toolbar to conditionally render based on whether the active tab is a home type, improving user experience.

limichange committed:
feat: Refactor web site handler to support active tab updates

- Removed unused state in Welcome component for cleaner code.
- Enhanced useWebSiteHandler to update the active tab's properties when navigating from a home type to a normal type, improving tab management and user experience.

limichange committed:
refactor: Simplify addBrowserHomeTab method and update WelcomeItem component

- Streamlined the addBrowserHomeTab method by removing unnecessary object spread syntax for improved readability.
- Refactored WelcomeItem component to utilize useWebSiteHandler for handling website actions, enhancing code clarity and maintainability.

limichange committed:
refactor: Clean up imports and streamline website handling in HistoryListModal

- Removed unused `useMedia` import and related code in HistoryListModal for improved clarity.
- Integrated `useWebSiteHandler` to manage website actions, enhancing code maintainability.
- Updated website handling logic to conditionally pop navigation based on user actions.

limichange committed:
Merge branch 'x' into OK-36161

limichange commented:
@rabbitcode resume

originalix commented on packages/kit/src/hooks/useReplaceTx.ts:
```diff
@@ -136,7 +136,6 @@ function useReplaceTx({
           onConfirm: async () => {
             // https://www.f2pool.com/user/tx-acc?from=onekey&txid={txid}
             handleOpenWebSite({
-              switchToMultiTabBrowser: gtMd,
```
这里不应该改变交易加速的跳转逻辑，否则在桌面端就只会跳转到发现页首页，而不是鱼池的 dapp

limichange committed:
refactor: Simplify MobileTabItem props and enhance DesktopCustomTabBar navigation logic

- Removed unused `tabBarStyle` prop from MobileTabItem for cleaner code.
- Updated DesktopCustomTabBar to conditionally handle navigation based on platform, improving user experience on desktop and mobile.

limichange committed:
refactor: Clean up MobileBrowserBottomBar component and streamline props usage

- Removed unused `setWebTabData` from MobileBrowserBottomBar for cleaner code.
- Updated function parameter in MobileBrowserBottomBar to use `_props` for clarity.
- Enhanced `useTakeScreenshot` function to use `_id` parameter, improving consistency.

limichange committed:
refactor: Enhance website navigation handling with multi-tab support

- Added `switchToMultiTabBrowser` parameter to improve navigation logic in `useReplaceTx` and `ContextJotaiActionsDiscovery`.
- Updated `useWebSiteHandler` to utilize media query for determining tab behavior, enhancing user experience across devices.
- Streamlined navigation logic to conditionally switch tabs based on the new parameter and platform environment.

limichange committed:
refactor: Update dependencies in useWebSiteHandler for improved tab management

- Enhanced the dependency array in useWebSiteHandler to include activeTab properties, ensuring accurate updates and reactivity.
- Improved overall functionality and responsiveness of the website handler in relation to active tab changes.

limichange committed:
refactor: Update useReplaceTx to streamline website navigation handling

- Adjusted the `onConfirm` function in useReplaceTx to ensure consistent handling of the `switchToMultiTabBrowser` parameter.
- Improved clarity by removing redundant code and enhancing the logic for opening the website in a new tab.

limichange committed:
refactor: Update useReplaceTx to include gtMd in dependency array

- Added gtMd to the dependency array of useReplaceTx to ensure proper reactivity when the multi-tab browser switch state changes.
- Enhanced the logic for opening websites by passing the switchToMultiTabBrowser parameter, improving navigation handling.

limichange committed:
refactor: Update BookmarkListModal to streamline website handling

- Removed unused Input import and related code for cleaner structure.
- Integrated useWebSiteHandler for improved website navigation logic.
- Updated handleWebSite function to enhance clarity and maintainability in handling bookmark actions.

limichange committed:
refactor: Simplify website handling in BookmarkListModal

- Introduced handleItemPress function to streamline website navigation logic.
- Replaced inline website handling code with handleItemPress for improved clarity and maintainability.
- Removed unused import for cleaner code structure.

limichange committed:
refactor: Streamline website handling in SearchModal

- Replaced inline website handling logic with useWebSiteHandler for improved clarity and maintainability.
- Removed unused imports and parameters to clean up the code structure.
- Enhanced navigation logic for better user experience when opening websites.

limichange committed:
refactor: Enhance SearchModal website handling with improved navigation logic

- Replaced inline website handling with handleOpenWebSite for better clarity and maintainability.
- Integrated useCurrentWindow and tabId parameters to enhance navigation options.
- Updated logging for dApp entries to improve tracking and analytics.
- Cleaned up dependencies in the useEffect hook for better performance.

limichange committed:
refactor: Add tabId parameter to useWebSiteHandler for enhanced navigation options

- Introduced tabId parameter in IHandleWebSiteParams to support better tab management.
- Updated useWebSiteHandler to utilize tabId, improving the flexibility of website handling.
- Enhanced overall navigation logic for a more seamless user experience.

limichange committed:
refactor: Update enterMethod values for improved clarity in navigation

- Changed enterMethod from EEnterMethod.dashboard to EEnterMethod.bookmark in BookmarksSection for better context.
- Updated enterMethod from EEnterMethod.dashboard to EEnterMethod.trending in TrendingSection to reflect accurate navigation.
- Refactored SearchModal to utilize useWebSiteHandler for handling website navigation, enhancing code clarity and maintainability.
- Removed deprecated handleOpenWebSite calls and streamlined website handling logic throughout SearchModal.

limichange committed:
Merge branch 'x' into OK-36161

limichange committed:
refactor: Simplify website handling across multiple components

- Removed the use of 'void' in handleWebSite calls to enhance clarity in website navigation logic.
- Updated multiple components including BookmarkListModal, Dashboard sections, and SearchModal for consistent handling of website actions.
- Improved code maintainability by streamlining the website handling process throughout the application.

limichange committed:
Merge branch 'x' into OK-36161
